array/builder.go

```diff
@@ -0,0 +1,157 @@
+package array
+
+import (
+	"github.com/apache/arrow/go/arrow/array"
+	"github.com/apache/arrow/go/arrow/memory"
+)
+
+type StringBuilder struct {
+	builder      *array.BinaryBuilder
+	mem          memory.Allocator
+	value        string
+	length       int
+	capacity     int
+	dataCapacity int
+	refCount     int
+}
```

Comment:
Can you add some comments here that describe why we have a non-generated builder for strings, and when it makes the switch over to the arrow builder?

Reply:
Sure.